specification/mediaservices/resource-manager/Microsoft.Media/stable/2021-11-01/AssetsAndAssetFilters.json

```diff
@@ -0,0 +1,2032 @@
+          }
+        ],
+        "x-ms-long-running-operation": true,
+        "x-ms-examples": {
+          "Delete a Track": {
+            "$ref": "examples/assetTracks-delete.json"
+          }
+        }
+      },
+      "patch": {
+        "summary": "Update an Track",
+        "description": "Updates an existing Track in the asset",
+        "operationId": "Tracks_Update",
+        "responses": {
+          "202": {
+            "description": "Accepted",
+            "headers": {
+              "Retry-After": {
+                "description": "The recommended number of seconds to wait before calling the URI specified in Azure-AsyncOperation.",
+                "type": "integer",
+                "format": "int32"
+              },
+              "Location": {
+                "description": "The URI to poll for completion status.",
+                "type": "string"
+              },
+              "Azure-AsyncOperation": {
+                "description": "The URI to poll for completion status.",
+                "type": "string"
+              }
+            },
+            "schema": {
+              "$ref": "#/definitions/AssetTrack"
+            }
+          },
+          "default": {
+            "description": "Detailed error information.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionId"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupName"
+          },
+          {
+            "$ref": "#/parameters/MediaServicesAccountName"
+          },
+          {
+            "name": "assetName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The Asset name."
+          },
+          {
+            "name": "trackName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The Asset Track name."
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/AssetTrack"
+            },
+            "description": "The request parameters"
+          },
+          {
+            "$ref": "#/parameters/ApiVersion"
+          }
+        ],
+        "x-ms-long-running-operation": true,
+        "x-ms-examples": {
+          "Update a Track": {
+            "$ref": "examples/assetTracks-update.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Media/mediaServices/{accountName}/assets/{assetName}/tracks/{trackName}/updateTrackData": {
+      "post": {
+        "summary": "Update the track data",
+        "description": "Update the track data.",
```

Comment:
what does this operation do?

Reply:
These are subtitle text tracks for the Video, customers can change an updated version of the track by calling this API and sending the subtitle data.

Reply:
This API doesn't take any input, where is the subtitle data provided by the caller?

Reply:
The subtitle file is saved in customer's AzureStorage blob container. This API is added in such situation:

1. User uploaded a subtitle file to blob
2. User calls our AssetTrack API with PUT to create a new track
3. User replace the subtitile file with new content, like fix typo in subtitle file.
4. User call this UpdateTrackData API to tell our service to refresh the subtitle file.

If user does not call the UpdateTrackData, we will always use the old subtitle file. In this API, we will re-index the new subtitle file.

Reply:
Please consider including information like this in the operation description.  These descirptions, etc... drive the REST API documentation

Reply:
added better description.